source/develop/release-management/features/virt/tpm-device.md

```diff
@@ -91,12 +91,26 @@ The options to deliver updated TPM data are similar to obtaining TPM data on VM
 
 - Send the updated data to Engine in events, current API call responses (stats, destroy) or using a new Vdsm API call for the given purpose.
 - Store the data in metadata, to be read and processed in Engine monitoring.
-- Attach the data to dumpxmls
+- Attach the data to dumpxmls.
 - Store the data to shared storage from Vdsm.
 - Let swtpm store the data directly to the shared storage.
 - Some combination of the options above.
 
-Different options are discussed in the following subsections.  We can probably assume that TPM data updates are not going to be very frequent.  We mustn't forget about reboots, snapshots, imports and exports.
+We can assume that TPM data updates are not going to be very frequent.  We mustn't forget about reboots, snapshots, imports and exports.
+
+After considering all the options discussed in the following subsections, the following solution was selected:
+
+- TPM data is stored in the Engine database, in a separate table.
+- A new parameter providing TPM data is added to VM.create call.
+- A new API call VM.getExternalData to retrieve TPM data is introduced.
+- Engine calls VM.getExternalData to update the data in the database after a VM is stopped but before it is destroyed.
+- If a VM is powered off rather than shut down from Engine or from within the guest, the latest available stable data is retrieved.  It may not correspond to the very final data, but this is always a risk with unclean shutdown.
```

Comment:
right, and we mentioned the possibility to change power-off so it won't un-define the domain to ensure we'll get the latest data - the question is whether or not we can do it it a backward compatible way (I think we can)

Reply:
Explained in new Future work section.